### PR TITLE
Make numeric values and operations default to mathematical values

### PR DESCRIPTION
**Pull request by ExE-Boss**
Fixes <https://github.com/engine262/engine262/issues/155>

### CODE REVIEW

File: src/abstract-ops/spec-types.mjs [hunk 1]
```diff
@@ -15,7 +17,19 @@ import {
...
+export function 𝔽(x) {
+  Assert(typeof x === 'number');
+  return new NumberValue(x);
+}
+
+// #ℤ
+export function ℤ(x) {
+  Assert(typeof x === 'bigint');
+  return new BigIntValue(x);
+}
```
Comment (devsnek) [inline src/abstract-ops/spec-types.mjs:32]: can these just use `F` and `Z` instead?

Comment (ExE-Boss) [reply src/abstract-ops/spec-types.mjs]: For the most part, yes, but there are a few cases where they need to be renamed due to conflicts with local variables named `F` (in `abstract‑ops/function-operations.mjs` and `abstract‑ops/regexp-objects.mjs`)

File: src/abstract-ops/function-operations.mjs [hunk 2]
```diff
@@ -28,13 +28,14 @@ import {
...
   IsExtensible,
-  IsInteger,
   MakeBasicObject,
   OrdinaryObjectCreate,
   OrdinaryCreateFromConstructor,
   ToObject,
+  isNonNegativeInteger,
   isStrictModeCode,
   Realm,
+  F as 𝔽,
```
Comment (devsnek) [inline src/abstract-ops/function-operations.mjs:38]: It is ok to alias this here, but it needs to be something easy to type.

**Comment by devsnek** [comment]
@ExE-Boss have you run the test suite on this?

**Comment by ExE-Boss** [comment]
@devsnek Yes, I did.

File: src/intrinsics/ArrayBufferPrototype.mjs [hunk 3]
```diff
@@ -19,12 +20,12 @@ function ArrayBufferProto_byteLength(args, { thisValue }) {
   }
   // 4. If IsDetachedBuffer(O) is true, return +0f.
```
Comment (ExE-Boss) [inline src/intrinsics/ArrayBufferPrototype.mjs:21]: @devsnek You merged this just as I was pushing a commit to fix this comment to:
```suggestion
  // 4. If IsDetachedBuffer(O) is true, return +0𝔽.
```

**Pull request merged**